backend/src/bin/consumer/main.rs

```diff
@@ -0,0 +1,356 @@
+use backend::req::pulsar::*;
+use futures::TryStreamExt;
+use lazy_static::lazy_static;
+use pulsar::{Consumer, Pulsar, SubType, TokioExecutor};
+use sea_orm::prelude::DateTimeWithTimeZone;
+use serde_json::json;
+use std::env;
+use tokio::time::sleep;
+use tokio::time::Duration;
+
+lazy_static! {
+    static ref TYPESENSE_API_KEY: String = {
+        env::var("TYPESENSE_API_KEY")
+            .ok()
+            .unwrap_or_else(|| "8Dz4jRrsBjYgdCD/VGP1bleph7oBThJr5IcF43l0U24=".to_string())
+    };
+    static ref TYPESENSE_ADDR: String = {
+        env::var("TYPESENSE_ADDR")
+            .ok()
+            .unwrap_or_else(|| "http://127.0.0.1:8108".to_string())
+    };
+}
+
+// fn log_init() {
+//     match log4rs::init_file("conf/log4rs.yml", Default::default()) {
+//         Ok(_) => (),
+//         Err(e) => panic!("Error initial logger: {}", e),
+//     }
+// }
+
+async fn create_typesense_collections() -> Result<(), reqwest::Error> {
+    //create typesense collections
+    let collection_burrows = json!({
+      "name": "burrows",
+      "fields": [
+        {"name": "burrow_id", "type": "int64"},
+        {"name": "title", "type": "string" },
+        {"name": "introduction", "type": "string"},
+        {"name": "update_time", "type": "string"}
+      ]
+    });
+    let collection_posts = json!({
+      "name": "posts",
+      "fields": [
+        {"name": "post_id", "type": "int64" },
+        {"name": "title", "type": "string" },
+        {"name": "burrow_id", "type": "int64" },
+        {"name": "update_time", "type": "string"},
+        {"name": "post_type", "type": "int32"},
+        {"name": "post_state", "type": "int32"},
+        {"name": "section", "type": "string[]"},
+        {"name": "tag", "type": "string[]"}
+      ]
+    });
+    let collection_replies = json!({
+      "name": "replies",
+      "fields": [
+        {"name": "reply_id", "type": "int32" },
+        {"name": "post_id", "type": "int64", "facet":true},
+        {"name": "burrow_id", "type": "int64"},
+        {"name": "update_time", "type": "string"},
+        {"name": "content", "type": "string"},
+        {"name": "reply_state", "type": "int32"}
+      ]
+    });
+    let client = reqwest::Client::new();
+    for each in [collection_burrows, collection_posts, collection_replies].iter() {
+        // TODO: match the status code of Response here, to see whether it is successfully created or is already created, or failed
+        match client.build_post("/collections").json(&each).send().await {
+            Ok(a) => match a.status().as_u16() {
+                201 => (),
+                400 => panic!(
+                    "Bad Request - The request could not be understood due to malformed syntax."
+                ),
+                401 => panic!("Unauthorized - Your API key is wrong."),
+                404 => panic!("Not Found - The requested resource is not found."),
+                409 => panic!("Conflict - When a resource already exists."),
```

Comment:
This should not panic. Imagine a case: you run this script twice, so you create the resource twice. This should not stop the program running: You have it created correctly!